txcache/txByHashMap_test.go

```diff
@@ -0,0 +1,167 @@
+package txcache
+
+import (
+	"math/big"
+	"strconv"
+	"sync"
+	"testing"
+
+	"github.com/stretchr/testify/require"
+)
+
+func Test_newTxByHashMap(t *testing.T) {
+	txByHashmap := newTxByHashMap(1)
+	require.NotNil(t, txByHashmap)
+}
+
+func createMockWrappedTx(txHash []byte) *WrappedTransaction {
+	return &WrappedTransaction{
+		Tx:     nil,
+		TxHash: txHash,
+	}
+}
+
+func createSliceMockWrappedTxs(txHashes [][]byte) []*WrappedTransaction {
+	wrappedTxs := make([]*WrappedTransaction, len(txHashes))
+	for i, txHash := range txHashes {
+		wrappedTxs[i] = createMockWrappedTx(txHash)
+	}
+
+	return wrappedTxs
+}
+
+func createMockTxHashes(numberOfTxs int) [][]byte {
+	txHashes := make([][]byte, numberOfTxs)
+	for i := 0; i < numberOfTxs; i++ {
+		txHashes[i] = []byte("txHash" + strconv.Itoa(i))
+	}
+
+	return txHashes
+}
+
+func addWrappedTxs(txByHash *txByHashMap, wrappedTxs []*WrappedTransaction, numberOfTxs int) {
+	var wg sync.WaitGroup
+	wg.Add(numberOfTxs)
+
+	for _, wrappedTx := range wrappedTxs {
+		wrappedTxCopy := wrappedTx
```

Comment:
This is a mere copy of the pointer. This is not copy-assignment. Thus, it's name is somehow misleading. Actually, can it be removed?